protocols/autonat/src/v2/client/handler/dial_request.rs

```diff
@@ -0,0 +1,332 @@
+            .outbound
+            .try_push(req.nonce, start_stream_handle(req, rx))
+            .is_err()
+        {
+            tracing::debug!("Dial request dropped, too many requests in flight");
+        }
+    }
+}
+
+impl ConnectionHandler for Handler {
+    type FromBehaviour = DialRequest;
+    type ToBehaviour = ToBehaviour;
+    type InboundProtocol = DeniedUpgrade;
+    type OutboundProtocol = ReadyUpgrade<StreamProtocol>;
+    type InboundOpenInfo = ();
+    type OutboundOpenInfo = ();
+
+    fn listen_protocol(&self) -> SubstreamProtocol<Self::InboundProtocol, Self::InboundOpenInfo> {
+        SubstreamProtocol::new(DeniedUpgrade, ())
+    }
+
+    fn poll(
+        &mut self,
+        cx: &mut Context<'_>,
+    ) -> Poll<
+        ConnectionHandlerEvent<Self::OutboundProtocol, Self::OutboundOpenInfo, Self::ToBehaviour>,
+    > {
+        if let Some(event) = self.queued_events.pop_front() {
+            return Poll::Ready(event);
+        }
+
+        match self.outbound.poll_unpin(cx) {
+            Poll::Ready((nonce, Ok(outcome))) => {
+                return Poll::Ready(ConnectionHandlerEvent::NotifyBehaviour(
+                    ToBehaviour::TestOutcome { nonce, outcome },
+                ))
+            }
+            Poll::Ready((nonce, Err(_))) => {
+                return Poll::Ready(ConnectionHandlerEvent::NotifyBehaviour(
+                    ToBehaviour::TestOutcome {
+                        nonce,
+                        outcome: Err(Error::Io(io::ErrorKind::TimedOut.into())),
+                    },
+                ));
+            }
+            Poll::Pending => {}
+        }
+
+        Poll::Pending
+    }
+
+    fn on_behaviour_event(&mut self, event: Self::FromBehaviour) {
+        self.perform_request(event);
+    }
+
+    fn on_connection_event(
+        &mut self,
+        event: ConnectionEvent<
+            Self::InboundProtocol,
+            Self::OutboundProtocol,
+            Self::InboundOpenInfo,
+            Self::OutboundOpenInfo,
+        >,
+    ) {
+        match event {
+            ConnectionEvent::DialUpgradeError(DialUpgradeError { error, .. }) => {
+                tracing::debug!("Dial request failed: {}", error);
+                match self.queued_streams.pop_front() {
+                    Some(stream_tx) => {
+                        let _ = stream_tx.send(Err(error));
+                    }
+                    None => {
+                        tracing::warn!(
+                            "Opened unexpected substream without a pending dial request"
+                        );
+                    }
+                }
+            }
+            ConnectionEvent::FullyNegotiatedOutbound(FullyNegotiatedOutbound {
+                protocol, ..
+            }) => match self.queued_streams.pop_front() {
+                Some(stream_tx) => {
+                    if stream_tx.send(Ok(protocol)).is_err() {
+                        tracing::warn!("Failed to send stream to dead handler");
```

Comment:
I don't think this should be a `warn!`, `debug!` is enough :)

Reply:
It is now debug!